cmd/mobynit/main.go

```diff
@@ -65,22 +73,48 @@ func mountContainer(containerID, graphDriver string) string {
 	return newRootPath
 }
 
-func main() {
-	// Any mounts done by initrd will be transfered in the new root
-	mounts, err := mount.GetMounts()
-
-	rawGraphDriver, err := ioutil.ReadFile("/current/boot/storage-driver")
+func getStorageDriverAndContainerID(layer_root string) (string, string) {
+	rawGraphDriver, err := ioutil.ReadFile(filepath.Join(layer_root, "/current/boot/storage-driver"))
 	if err != nil {
 		log.Fatal("could not get storage driver:", err)
 	}
 	graphDriver := strings.TrimSpace(string(rawGraphDriver))
 
-	current, err := os.Readlink("/current")
+	current, err := os.Readlink(filepath.Join(layer_root, "/current"))
 	if err != nil {
 		log.Fatal("could not get container ID:", err)
 	}
 	containerID := filepath.Base(current)
 
+	return graphDriver, containerID
+}
+
+func main() {
+
+	layerPtr := flag.String("layer-root", "", "root of partition e.g. /mnt/sysroot/inactive. Mount destination is returned in stdout")
+	flag.Parse()
+
+	// Any mounts done by initrd will be transfered in the new root
+	mounts, err := mount.GetMounts()
+	if err != nil {
+		log.Fatal("could not get mounts:", err)
+	}
+
+	var graphDriver, containerID string
+
+	// If a custom layer-root is passed, use it instead of LAYER_ROOT
+	if *layerPtr != "" {
+		graphDriver, containerID = getStorageDriverAndContainerID(*layerPtr)
+		newRootPath := mountStoreFromOptions(filepath.Join(*layerPtr, "/balena"), containerID, graphDriver)
+		if newRootPath != "" {
+			os.Stdout.WriteString(newRootPath)
```

Comment:
you should use `fmt.Println(newRootPath)` or `log.Println(newRootPath)` to print things